async.go

```diff
@@ -148,25 +148,25 @@ func run(ctx context.Context, fn func(context.Context) error) error {
 }
 
 func execute(ctx context.Context, task func(context.Context) error, taskName string, logger log.Logger) error {
-	ctx = getContextWithTaskName(ctx, taskName)
+	newCtx := getContextWithTaskName(ctx, taskName)
```

Comment:
you can just use `ctx` itself here,

Reply:
we are adding taskname in ctx. if we use existing ctx it will override taskname value in case of running multiple tasks